cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -678,15 +678,31 @@ class HydratorPlusPlusTopPanelCtrl {
     };
     this.previewLoading = true;
     this.loadingLabel = 'Stopping';
-    this.stopTimer();
-    this.updateTimerLabelAndTitle();
     this.myPipelineApi
         .stopPreview(params, {})
         .$promise
         .then(
-          (err) => {
+          () => {
+            this.stopTimer();
+            this.updateTimerLabelAndTitle();
             this.previewLoading = false;
             this.previewRunning = false;
+            this.dataSrc.stopPoll(this.pollId);
+            this.pollId = null;
+
+            let pipelinePreviewPlaceholder = 'The preview of the pipeline';
+            let pipelineName = this.HydratorPlusPlusConfigStore.getName();
```

Comment:
Can we use `const` for `pipelineName`, since its values are not changed?

Reply:
Fixed!